cdci_data_analysis/flask_app/dispatcher_query.py

```diff
@@ -570,12 +569,15 @@ def run_call_back(self, status_kw_name='action') -> typing.Tuple[str, typing.Uni
         try:
             if self.is_email_to_send_callback(status, time_original_request):
 
-                # build the products URL
-                products_url = self.generate_products_url_from_file(self.config.products_url, session_id, self.job_id)
+                # build the products URL and get also the original requested product
+                original_request_par_dic = self.get_request_par_dic()
+                products_url = self.generate_products_url_from_file(self.config.products_url, request_par_dict=original_request_par_dic)
+                product_type = original_request_par_dic.get('product_type', '')
```

Comment:
Can it really be empty? If it is, it is probably an issue, and it should cause an exception (to sentry).

Reply:
Actually, I realized this was already taken care of few lines below.